client/db.go

```diff
@@ -109,7 +110,16 @@ type Store interface {
 	//
 	// Field [FieldKind] values may be provided in either their raw integer form, or as string as per
 	// [FieldKindStringToEnumMapping].
-	PatchSchema(context.Context, string) error
+	PatchSchema(context.Context, string, bool) error
+
+	// SetDefaultSchemaVersion sets the default schema version to the ID provided.  It will be applied to all
+	// collections using the schema.
+	//
+	// This will affect all operations interacting with the schema where a schema version is not explicitly
+	// provided.  This includes GQL queries and Collection operations.
+	//
+	// It will return an error if the provided schema version ID does not exist.
+	SetDefaultSchemaVersion(context.Context, string) error
```

Comment:
This was confusing for me at first. What do you think about moving this to the Collection interface instead? I think it  would be more clear exactly which Collection you are setting a default for.

Reply:
The simplicity is nice but I can see many users getting confused. 

This is the use case I see being used most often when you don't know the schema version id:

```go
cols := db.GetAllCollections()

for _, col := range cols {
  if isMatchingCollection(col) {
    col.SetDefaultSchemaVersion()
  }
}

```

Reply:
I do disagree :) I see the most common use case to within a migration script:

```
defradb patchSchema jsonStuff
...
defradb setDefaultSchemaVersion hjndsmakdgamkmhga
```
if it was on the collection object it would be a bit stranger no, would look something like the below no?
```
defradb collection hjndsmakdgamkmhga setDefaultSchemaVersion
```

Reply:
It's definitely more verbose but it's also much easier to parse what that command is doing. 

No need to block the PR. I think we could add the shortcut function in the future if users have trouble with it.